module/zfs/metaslab.c

```diff
@@ -752,7 +938,7 @@ metaslab_group_alloc_update(metaslab_group_t *mg)
 	was_allocatable = mg->mg_allocatable;
 	was_initialized = mg->mg_initialized;
 
-	mg->mg_free_capacity = ((vs->vs_space - vs->vs_alloc) * 100) /
+	uint64_t free_capacity = ((vs->vs_space - vs->vs_alloc) * 100) /
```

Comment:
Should `vd->vdev_stat_lock` be held here?  This is unrelated to your PR, as the existing code is not checking for the lock, but thought I'd mention.

Reply:
Same as above, I think no.